work_queue/src/work_queue.h

```diff
@@ -346,6 +346,38 @@ struct work_queue_stats {
 	int workers_blacklisted;         /**< @deprecated Use workers_blocked instead. */
 };
 
+/** Statistics describing the workers in a work queue. Workers are grouped by index. For example, count[0] contains the number of workers of type 0, cores[0] contains the number of cores each of those workers has, memory[0] displays the memory of those workers, etc**/
+
+struct work_queue_wsummary {
```

Comment:
Hmm, the fixed array sizes are going to be a problem at some point.

I would suggest something more like this.  Let the structure describe one bucket:

```
struct work_queue_wsummary {  int count; int cores; int memory; ... };
```

Then, let the user indicate the length of array they are willing to deal with:

```
work_queue_summary( struct work_queue *q, struct work_queue_wsummary [], int length ); 
```

So then it would get invoked like this:

```
struct work_queue_wsummary summary[64];
work_queue_summary(queue,summary,64);
```

And that way, the library will work with any arbitrary length.